docs/source/1.0/spec/core/behavior-traits.rst

```diff
@@ -655,3 +655,95 @@ See
             input: PutSomethingInput,
             output: PutSomethingOutput
         }
+
+
+.. _httpChecksum-trait:
+
+----------------------
+``httpChecksum`` trait
+----------------------
+
+Summary
+    Indicates that an operation's HTTP Request or Response supports checksum
+    validation. At least one of request or response checksum properties must
+    be specified within the trait.
+Trait selector
+    ``operation``
+Value type
+    ``structure``
+
+The `httpChecksum` trait is a structure that contains the following members:
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 10 80
+
+    * - Property
+      - Type
+      - Description
+    * - request
+      - ``[`` :ref:`HttpChecksumProperties structure <checksum-properties>` ``]``
+      - The `request` structure values define checksum validation behavior for
+        HTTP Request.
+
+    * - response
+      - ``[`` :ref:`HttpChecksumProperties structure <checksum-properties>` ``]``
+      - The `response` structure values define checksum validation behavior for
+        HTTP Response.
+
+
+.. - checksum-properties:
+
+HTTPChecksumProperties structure
+================================
+
+HTTPChecksumProperties defines checksum validation behavior using the
+following members:
+
+.. list-table::
+    :header-rows: 1
+    :widths: 10 10 80
+
+    * - Property
+      - Type
+      - Description
+    * - prefix
+      - ``string``
+      - **Required**. The prefix string is used to construct a header for a
+        checksum type. Prefix is a required member, and SHOULD be lower-cased. The
+        recommended ABNF to follow for prefix is `([a-z]*([a-z]/[0-9]/"-")*`.
+
+    * - location
```

Comment:
Is there a reason this property isn't an enum when the valid locations are "header" and "trailer"?

Reply:
No reason, good call out, will define a LOCATION enum.